### DISCUSSION TITLE
What should be the value of `Authorized JavaScript Origins` while creating google OAuth2 Client?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#3962] [Q&A] [answered]
Hello there 👋, I want to use google OAuth2 service to let users login. However, google asks to provide it with authorized origins. I used `http://127.0.0.1:5173` as I'm using sveltekit in dev mode. However, this seems to be not working. Are there any restrictions I must follow while setting the authorized origins?

**Comment by ganigeorgiev** [comment]
I believe you can leave it empty as the OAuth2 API interactions are performed on the server-side.

But note that you still have to specify the redirect uri(s) under the "Authorized redirect URIs" section.

**Reply by Mr-M1M3** [reply]
If I bind a value to it, are there any chances of pb to respond with a status of 400 saying, `something went wrong while processing your request`?

**Reply by ganigeorgiev** [reply]
No, I don't think so.
Make sure to check your `users` collection Create API rule and validation constraints.

You can try to find the failed request in the _Admin UI > Logs_ section and see if there is a more detailed error payload once you click on the failed log.

**Reply by Mr-M1M3** [reply]
Log says:
```
oauth2: "invalid_grant" "Malformed auth code."
```
What should I do to fix it?
**_I'm not using pb sdk. Using HTTPie, an API client._**

**Reply by ganigeorgiev** [reply]
There are really not enough details to help you debug it and I don't think your issue is PocketBase related.
If you need further help please elaborate and provide a minimal code sample showing how you are authenticating and what requests you are sending.

**Reply by Mr-M1M3** [reply]
I'm just messing around google oauth2 with pocketbase. However, I'm not using any sdk of pocketbase.

- ### After creating oauth client and setting up on pocketbase, I opened up HTTPie (an api client just like postman) and requested:
`http://localhost:8090/api/collections/users/auth-methods`
In Response I got:
```json
{
  "authProviders": [
    {
      "name": "google",
      "displayName": "Google",
      "state": "y1ywuJ4MFGn9ztfamTSjfkx0NcuG7g",
      "authUrl": "https://accounts.google.com/o/oauth2/auth?client_id=549095646206-fbj551sqrgjbfmd8m8miuj254dar94et.apps.googleusercontent.com&code_challenge=np_VExoH1P06d7WnECgAhf4VQ2csWrjM4VymJq3ebh8&code_challenge_method=S256&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&state=y1ywuJ4MFGn9ztfamTSjfkx0NcuG7g&redirect_uri=",
      "codeVerifier": "B4KyIMjoKywBae0zlZpOVr3i2hk9jJ10f5OjzvNPwuA",
      "codeChallenge": "np_VExoH1P06d7WnECgAhf4VQ2csWrjM4VymJq3ebh8",
      "codeChallengeMethod": "S256"
    }
  ],
  "usernamePassword": false,
  "emailPassword": false,
  "onlyVerified": false
}
```
- ### Now, I copied the value of `authUrl` property I got back from pocketbase and added appropriate value for `redirect_uri` query param and then visited the url via my browser.
- ### After getting redirected, I copied the value of `code` query param:
`4%2F0AfJohXli62hFcjmhxn7EKUdm5zivvgZZlNrKjnrK7zmFdgfoxYHLTLiVEADuvk3xFgL2OA`
- ### On HTTpie, I copied the value of `codeVerifier` from previous response and made a post request to: 
`http://localhost:8090/api/collections/users/auth-with-oauth2`
with data: 
```json
{
"provider": "google",
"code":"4%2F0AfJohXnGlx5A0J_iqQbmi3Maca33JulaDeSwmS5EVNIDL9ZsMIC5m8bEg6sjGAaaX2qgIg",
"codeVerifier":"touHxOtaV0shKMUGrJr5rkr2tbxR1Il91Xcq6f9E4hI",
 "redirectUrl": "http://localhost:5173"
}
```
In response, I got
```json
{
  "code": 400,
  "message": "Something went wrong while processing your request.",
  "data": {}
}
```

**Reply by ganigeorgiev** [reply, marked as answer]
The `4%2F` part looks like it is either from some other query param and you didn't copied it properly and/or you may need to url decode the code value before sending it as part of the request body.

**Reply by Mr-M1M3** [reply]
Thanks A Lottt..... URI decoding seems to be working.
